designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Dtos.scala

```diff
@@ -0,0 +1,709 @@
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List.empty,
+    )
+
+    // Scenario modifications
+
+    def forScenarioModified(
+        id: String,
+        user: String,
+        date: Instant,
+        scenarioVersion: Long,
+        comment: Option[String],
+    ): ScenarioActivity = ScenarioActivity(
+      id = id,
+      `type` = ScenarioActivityType.ScenarioModified,
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List.empty,
+      overrideDisplayableName = Some(s"Version $scenarioVersion saved"),
+    )
+
+    def forScenarioNameChanged(
+        id: String,
+        user: String,
+        date: Instant,
+        scenarioVersion: Long,
+        comment: Option[String],
+        oldName: String,
+        newName: String,
+    ): ScenarioActivity = ScenarioActivity(
+      id = id,
+      `type` = ScenarioActivityType.ScenarioNameChanged,
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List(
+        AdditionalField("oldName", oldName),
+        AdditionalField("newName", newName),
+      )
+    )
+
+    def forCommentAdded(
+        id: String,
+        user: String,
+        date: Instant,
+        scenarioVersion: Long,
+        comment: Option[String],
+    ): ScenarioActivity = ScenarioActivity(
+      id = id,
+      `type` = ScenarioActivityType.CommentAdded,
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List.empty,
+    )
+
+    def forCommentAddedAndDeleted(
+        id: String,
+        user: String,
+        date: Instant,
+        scenarioVersion: Long,
+        comment: Option[String],
+        deletedByUser: String,
+    ): ScenarioActivity = ScenarioActivity(
+      id = id,
+      `type` = ScenarioActivityType.CommentAdded,
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List(
+        AdditionalField("deletedByUser", deletedByUser),
+      ),
+      overrideSupportedActions = Some(List.empty)
+    )
+
+    def forAttachmentAdded(
+        id: String,
+        user: String,
+        date: Instant,
+        scenarioVersion: Long,
+        comment: Option[String],
+        attachmentId: String,
+        attachmentFilename: String,
+    ): ScenarioActivity = ScenarioActivity(
+      id = id,
+      `type` = ScenarioActivityType.AttachmentAdded,
+      user = user,
+      date = date,
+      scenarioVersion = scenarioVersion,
+      comment = comment,
+      additionalFields = List(
+        AdditionalField("attachmentId", attachmentId),
+        AdditionalField("attachmentFilename", attachmentFilename),
+      )
+    )
+
+    def forAttachmentAddedAndDeleted(
```

Comment:
The name is strange for me because how an attachment can be added and deleted at once. The same action is assigned to add and delete and it's controlled by an additional field `deleteByUser`?

Reply:
refactored, forAttachmentPresent/forAttachmentDeleted, but it may slightly change after full BE implementation